File: src/main/java/no/hvl/tk/visual/debugger/debugging/visualization/DebuggingInfoVisualizer.java
package no.hvl.tk.visual.debugger.debugging.visualization;

import com.intellij.debugger.jdi.StackFrameProxyImpl;
import no.hvl.tk.visual.debugger.debugging.stackframe.StackFrameAnalyzer;
import no.hvl.tk.visual.debugger.domain.ObjectDiagram;

public interface DebuggingInfoVisualizer {

  void doVisualization(ObjectDiagram diagram);

  void reprintDiagram();

  void addMetadata(String fileName, Integer line, StackFrameAnalyzer stackFrame);

  ObjectDiagram getObjectWithChildren(String objectId);

  void debuggingActivated();

  void debuggingDeactivated();

  void sessionStopped();
}

File: src/main/java/no/hvl/tk/visual/debugger/debugging/visualization/DebuggingInfoVisualizer.kt
package no.hvl.tk.visual.debugger.debugging.visualization

import no.hvl.tk.visual.debugger.debugging.stackframe.StackFrameAnalyzer
import no.hvl.tk.visual.debugger.domain.ObjectDiagram

interface DebuggingInfoVisualizer {
  fun doVisualization(diagram: ObjectDiagram)

  fun reprintDiagram()

  fun addMetadata(fileName: String, line: Int, stackFrameAnalyzer: StackFrameAnalyzer)

  fun getObjectWithChildren(objectId: String): ObjectDiagram

  fun debuggingActivated()

  fun debuggingDeactivated()

  fun sessionStopped()
}
